src/service/xhr-impl.js

```diff
@@ -189,8 +203,39 @@ export class Xhr {
       init.headers['Content-Type'] = 'application/json;charset=utf-8';
       init.body = JSON.stringify(init.body);
     }
-    return this.fetch(input, init)
-        .then(response => response.json());
+
+    const isCacheable = (init.method === 'GET' && propertyPath);
+    if (isCacheable) {
+      const cacheKey = this.getCacheKey_(input, opt_init);
+      const cachedPromise = this.fragmentCache_.get(cacheKey);
```

Comment:
Hmm I'm not sure I understand. The fragment is a path to a property inside a JSON object. How should fetchDocument or fetchText specify a subset of data like the issue specs?